testpar/t_cache.c

```diff
@@ -6938,6 +6938,26 @@ main(int argc, char **argv)
         goto finish;
     }
 
+    if (NULL == (data = HDmalloc(NUM_DATA_ENTRIES * sizeof(*data)))) {
+        if (MAINPROCESS)
+            HDprintf("    Couldn't allocate data array.  Exiting.\n");
+        MPI_Abort(MPI_COMM_WORLD, -1);
+    }
+    if (NULL == (data_index = HDmalloc(NUM_DATA_ENTRIES * sizeof(*data_index)))) {
+        if (MAINPROCESS)
+            HDprintf("    Couldn't allocate data index array.  Exiting.\n");
+        MPI_Abort(MPI_COMM_WORLD, -1);
+    }
+
+    HDmemset(filenames, 0, sizeof(filenames));
+    for (int i = 0; i < NFILENAME; i++) {
+        if (NULL == (filenames[i] = HDmalloc(PATH_MAX))) {
+            if (MAINPROCESS)
+                HDprintf("couldn't allocate filename array\n");
```

Comment:
If anything other than rank 0 can't allocate memory, the program will exit without an error message.

Reply:
On the other hand, if they all can't allocate memory, then stdout will blow up